relay/sources/relayd/src/data/node.rs

```diff
@@ -123,6 +123,10 @@ impl NodesList {
         Ok(NodesList { list: nodes, my_id })
     }
 
+    pub fn get_nodeslist(&self) -> &NodesList {
```

Comment:
The idea of the fields not being public is to prevent code from outside of this module to rely on the internal representation of the nodes list, which may evolve. Instead you should create a dedicated method for what you are trying to to. In this case, it is even easier is it seems to already be there :)